Editor/Modules/FogModeAnalyzer.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using Unity.ProjectAuditor.Editor.Core;
+using Unity.ProjectAuditor.Editor.Diagnostic;
+using UnityEditor;
+using UnityEngine.Rendering;
+
+namespace Unity.ProjectAuditor.Editor.Modules
+{
+    public enum FogMode
+    {
+        Linear,
+        Exponential,
+        ExponentialSquarred,
+        Automatic
+    }
+
+    class FogModeAnalyzer : ISettingsModuleAnalyzer
+    {
+        private static readonly Descriptor k_FogModeDescriptor = new Descriptor(
+            "PAS1003",
+            "Graphics: Fog Shader Variant Stripping",
+            new[] {Area.BuildSize},
+            "FogMode shader variants are always built. Forcing Fog shader variants to be built can increase the build size.",
+            "To reduce the number of shader variants, change <b>Edit ➔ Project Settings ➔ Graphics ➔ Fog Modes</b> to <b>Automatic</b> or disable <b>Linear/Exponential/Exponential Squared</b>.")
+        {
+            fixer = (issue =>
+            {
+                RemoveFogStripping();
+            }),
+
+            messageFormat = "Graphics: FogMode {0} shader variants are always included in the build."
+        };
+
+        public void Initialize(ProjectAuditorModule module)
+        {
+            module.RegisterDescriptor(k_FogModeDescriptor);
+        }
+
+        public IEnumerable<ProjectIssue> Analyze(ProjectAuditorParams projectAuditorParams)
+        {
+            if (!IsFogStrippingEnabled(FogMode.Automatic))
```

Comment:
I think it would be better to report 3 different diagnostics. It's going to simplify a dew things too.
```
if (IsFogStrippingEnabled(FogMode.Linear))
{
                yield return ProjectIssue.Create(IssueCategory.ProjectSetting, k_FogModeDescriptor, FogMode.Linear)
                    .WithLocation("Project/Graphics");
}
if (IsFogStrippingEnabled(FogMode.Squared))
{
                yield return ProjectIssue.Create(IssueCategory.ProjectSetting, k_FogModeDescriptor, FogMode.Squared)
                    .WithLocation("Project/Graphics");
}
if (IsFogStrippingEnabled(FogMode.Exponential))
{
                yield return ProjectIssue.Create(IssueCategory.ProjectSetting, k_FogModeDescriptor, FogMode.Exponential)
                    .WithLocation("Project/Graphics");
}
```

Reply:
Why so? Isn't the bool check the easiest way to do that?

Reply:
The idea here is that we report 3 different messages. `ProjectIssue.Create` will take care of the formatting.